src/mixings/base_mixing.h

```diff
@@ -12,19 +12,14 @@
 //! Abstract base class for a generic mixture model
 
 //! This class represents a mixture model object to be used in a BNP iterative
-//! algorithm. By definition, a mixture is a probability distribution that
-//! integrates over a density kernel to generate the actual distribution for
-//! the data. However, in the context of this library, where a clustering
-//! structure is generated on the data, a certain mixture translates to a
-//! certain way of weighing the insertion of data in old clusters vs the
-//! creation of new clusters. Therefore any mixture object inheriting from the
-//! class must have methods that provide the probabilities for the two
-//! aforementioned events. The class will then have its own parameters, and
-//! maybe even prior distributions on them.
+//! algorithm.
 
 class BaseMixing {
  protected:
+  //!
   std::shared_ptr<google::protobuf::Message> prior;
+  //!
+  Eigen::MatrixXd *covariates_ptr;
```

Comment:
Why not use a constant reference? It should be safer imo

Reply:
Beacuse you said "pointer" and I mindlessly did just that 🙃. But you're absolutely correct.
By the way, should its name be `covariates_ptr` or just `covariates`? I thought I would make clear that it is not the actual matrix.

Reply:
Oh wait, I just realized using a const ref is impossible since it needs to receive a value as soon as it is created. But a pointer to a const matrix will do just fine.

Reply:
damn those const references are sneaky, good catch! pointer to const object looks great!